feature/qos/otg_tests/egress_strict_priority_scheduler_test/egress_strict_priority_scheduler_test.go

```diff
@@ -0,0 +1,2576 @@
+		setWeight:   false,
+		priority:    oc.Scheduler_Priority_STRICT,
+		inputID:     "AF3",
+		inputType:   oc.Input_InputType_QUEUE,
+		queueName:   queues.AF3,
+		targetGroup: "target-group-AF3",
+	}, {
+		desc:        "scheduler-policy-AF4",
+		sequence:    uint32(0),
+		setPriority: true,
+		setWeight:   false,
+		priority:    oc.Scheduler_Priority_STRICT,
+		inputID:     "AF4",
+		inputType:   oc.Input_InputType_QUEUE,
+		queueName:   queues.AF4,
+		targetGroup: "target-group-AF4",
+	}, {
+		desc:        "scheduler-policy-NC1",
+		sequence:    uint32(0),
+		setPriority: true,
+		setWeight:   false,
+		priority:    oc.Scheduler_Priority_STRICT,
+		inputID:     "NC1",
+		inputType:   oc.Input_InputType_QUEUE,
+		queueName:   queues.NC1,
+		targetGroup: "target-group-NC1",
+	}}
+
+	schedulerPolicy := q.GetOrCreateSchedulerPolicy("scheduler")
+	schedulerPolicy.SetName("scheduler")
+	t.Logf("QoS scheduler policies config: %v", schedulerPolicies)
+	for _, tc := range schedulerPolicies {
+		s := schedulerPolicy.GetOrCreateScheduler(tc.sequence)
+		s.SetSequence(tc.sequence)
+		if tc.setPriority {
+			s.SetPriority(tc.priority)
+		}
+		input := s.GetOrCreateInput(tc.inputID)
+		input.SetId(tc.inputID)
+		input.SetInputType(tc.inputType)
+		input.SetQueue(tc.queueName)
+		gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
+	}
+
+	t.Logf("Create QoS output interface config")
+	schedulerIntfs := []struct {
+		desc      string
+		queueName string
+		scheduler string
+	}{{
+		desc:      "output-interface-BE1",
+		queueName: queues.BE1,
+		scheduler: "scheduler",
+	}, {
+		desc:      "output-interface-AF1",
+		queueName: queues.AF1,
+		scheduler: "scheduler",
+	}, {
+		desc:      "output-interface-AF2",
+		queueName: queues.AF2,
+		scheduler: "scheduler",
+	}, {
+		desc:      "output-interface-AF3",
+		queueName: queues.AF3,
+		scheduler: "scheduler",
+	}, {
+		desc:      "output-interface-AF4",
+		queueName: queues.AF4,
+		scheduler: "scheduler",
+	}, {
+		desc:      "output-interface-NC1",
+		queueName: queues.NC1,
+		scheduler: "scheduler",
+	}}
+
+	t.Logf("QoS output interface config: %v", schedulerIntfs)
+	for _, tc := range schedulerIntfs {
+		i := q.GetOrCreateInterface(dp3.Name())
+		i.SetInterfaceId(dp3.Name())
+		i.GetOrCreateInterfaceRef().Interface = ygot.String(dp3.Name())
+		if deviations.InterfaceRefConfigUnsupported(dut) {
+			i.InterfaceRef = nil
+		}
+		output := i.GetOrCreateOutput()
+		schedulerPolicy := output.GetOrCreateSchedulerPolicy()
+		schedulerPolicy.SetName(tc.scheduler)
+		queue := output.GetOrCreateQueue(tc.queueName)
+		queue.SetName(tc.queueName)
+		gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
+	}
+}
+
+func ConfigureDUTQoSMPLS(t *testing.T, dut *ondatra.DUTDevice) {
+	t.Helper()
+	dp1 := dut.Port(t, "port1")
+	dp2 := dut.Port(t, "port2")
+	dp3 := dut.Port(t, "port3")
+	d := &oc.Root{}
+	q := d.GetOrCreateQos()
+	queues := netutil.CommonTrafficQueues(t, dut)
+
+	if deviations.QOSQueueRequiresID(dut) {
```

Comment:
this deviation should be moved into the cfgplugin for generating qos queue config (probably called NewQoSQueue)